analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs

```diff
@@ -91,8 +91,8 @@ protected override void Initialize(SonarAnalysisContext context)
                 c =>
                 {
                     var stringLiteral = (TLiteralExpressionSyntax)c.Node;
-                    if (UriRegex.IsMatch(GetLiteralText(stringLiteral)) &&
-                        IsInCheckedContext(stringLiteral, c.SemanticModel))
+                    if (UriRegex.IsMatch(GetLiteralText(stringLiteral))
```

Comment:
I would merge the nongeneric and generic base classes. The nongeneric doesn't seem to have practical value

Reply:
I will let this for another time.